src/LibraryManager/Providers/Cdnjs/CdnjsProvider.cs

```diff
@@ -93,6 +92,27 @@ public ILibraryCatalog GetCatalog()
 
             desiredState = updateResult.InstallationState;
 
+            ILibraryCatalog catalog = GetCatalog();
```

Comment:
This new entire block should be moved out to another method (ValidateFilesInLibrary ??). All these "pre-restore" validation (validation of library Properties (id, destination, provider, etc), validation of requested Files in the files Property and conflicts detection) could potentially be part of the IProvider interface and no op for FileSystemProvider for things that do not apply (e.g.: files in the Files properties since we support rename).

Reply:
Actually UpdateStateAsync could just do this work. Right now it just expands the files set if not explicitly specified. But we can add an extra step to validate the explicitly listed files list against the catalog.

Reply:
removed it all and added a call in the UpdateStateAsync method to library.GetInvalidFiles(desiredState.Files);